tests/concurrency.py

```diff
@@ -0,0 +1,19 @@
+"""
+This module contains concurrency utilities that are only used in tests, thus not
+required as part of the ConcurrencyBackend API.
+"""
+
+import asyncio
+import functools
+
+from httpx import AsyncioBackend
+
+
+@functools.singledispatch
+async def sleep(backend, seconds: int):
+    raise NotImplementedError  # pragma: no cover
+
+
+@sleep.register(AsyncioBackend)
+async def _sleep_asyncio(backend, seconds: int):
+    await asyncio.sleep(seconds)
```

Comment:
@sethmlarson @tomchristie Thoughts on this pattern?

The HTTP/2 mock dispatcher `asyncio.sleep()` but we didn't need it anywhere else — a similar situation checking the usage of TLS in #263.

I figured there should be a way to gather tests-only concurrency utilities without having to reimplement a backend; a `singledispatch` on the backend type seemed like a good fit.

Reply:
I haven't seen this pattern before, it looks good though :)